app/interactors/access_limit/update_interactor.rb

```diff
@@ -28,14 +28,18 @@ def find_and_lock_edition
   def update_access_limit
     limit_type = params.require(:limit_type)
 
-    if LIMIT_TYPES.include?(limit_type)
-      context.fail! if edition.access_limit&.limit_type == limit_type
-      edition.assign_access_limit(limit_type, user)
+    if LIMIT_TYPES.exclude?(limit_type)
+      context.fail! if edition.access_limit.nil?
+      edition.assign_as_edit(user, access_limit: nil)
       return
     end
 
-    context.fail! if edition.access_limit.nil?
-    edition.remove_access_limit(user)
+    access_limit = AccessLimit.build_initial(created_by: user,
+                                             edition: edition,
+                                             limit_type: limit_type)
```

Comment:
If this is the only place we're calling this method and we're only saving one keyword argument from creating the model directly we may as well just call new rather than build_initial - particularly as we don't have a precedent for a `build_initial` pattern

Reply:
Indeed we could. The problem I had was trying to make sure it's covered and wanting to avoid writing a spec for the entire interactor to do so. It's a tenuous one because the attribute is purely archival. Would a spec for the interactor seem surprising?

Reply:
I'd have thought a spec for the interactor would be fine - we already have a few and they seem a good practice. I'm also not massively concerned about this particular thing being covered since it's enforced at a db level so only thing we can really test is that the expected revision made it as an association.

Reply:
Oh yeah, it is enforced - I mixed that up with the 'created_by_id'. In that case, I believe we're reasonably covered and don't see the motivation for additional testing of the interactor. Is that something you feel is important here? Anyway, I've removed the `build_initial` method 👍 .